### PR TITLE
Enable/Disable an Icinga2 Service

### PR DESCRIPTION
**Pull request by smbambling**
Similar to Pull #11 

This allows you to specify an array for enabled and disabled features via a parameter under the icinga2::server class.

The features.pp class will check to see if a feature has been added in both arrays and only disable it if it has been.

The enable.pp define will check to make sure the *.conf file exists in features-available before trying to enable it and checks to make sure a file/symlink is created in features-enabled to make sure it does exec multiple times.

Additionally a new object manifest has been added to set the conf file for graphite (GraphiteWriter).

### CODE REVIEW

File: manifests/object/graphitewriter.pp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Configure GraphiteWriter Feature .conf file
+define icinga2::object::graphitewriter (
+  $host = '127.0.0.1',
```
Comment (nickchappell) [inline manifests/object/graphitewriter.pp]: Could you change these names to `graphite_host` and `graphite_port`? Sorry to nitpick, but I want to keep parameter names descriptive, as the module will get tons of parameters as it grows in size.

**Comment by nickchappell** [comment]
Hi @smbambling,

Would you be able to submit the features-enabled changes and the GraphiteWriter changes as separate PRs?

File: templates/object_graphitewriter.conf.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ * WARNING: This IdoPgsqlConnection definition is automatically generated by Puppet.
```
Comment (nickchappell) [inline templates/object_graphitewriter.conf.erb]: Typo here.

File: manifests/object/graphitewriter.pp (hunk 1 above)
Comment (smbambling) [inline manifests/object/graphitewriter.pp]: Sure thing

**Comment by smbambling** [comment]
Sure I'll split these into two PRs.

File: manifests/object/graphitewriter.pp (hunk 1 above)
Comment (smbambling) [inline manifests/object/graphitewriter.pp]: Just to note the icinga2::object::idomysqlconnection and icinga2::object::idopgsqlconnection defines have non-descriptive parameters.  Ex. host,port,user ....etc.

Comment (nickchappell) [inline manifests/object/graphitewriter.pp]: Noted. I should have time this weekend to clean most of the smaller stuff up and to test/merge these PRs.

**Comment by nickchappell** [comment]
Hi @smbambling,

I tested this functionality and it looks good, but when I applied the patch, it still brought in the GraphiteWriter commits. Would you be able to rebase and repush this PR against the current master branch with just the commits for the feature enable/disable code?

**Comment by nickchappell** [comment]
Hi @smbambling,

I figured out what was going on and merged this PR in. Thanks!

**Pull request closed without merging**